stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -245,121 +245,97 @@ def apply_collections_filter(search: Search, collection_ids: List[str]):
+                - A datetime range string (e.g., "2023-01-01/2023-12-31")
+                - A datetime object
+                - A tuple of (start_datetime, end_datetime)
 
         Returns:
-            Search: The filtered search object.
+            The filtered search object.
         """
+        if not interval:
+            return search
+
         should = []
-        datetime_search = return_date(interval)
+        try:
+            datetime_search = return_date(interval)
+        except (ValueError, TypeError) as e:
+            # Handle invalid interval formats if return_date fails
+            logger.error(f"Invalid interval format: {interval}, error: {e}")
+            return search
 
-        # If the request is a single datetime return
-        # items with datetimes equal to the requested datetime OR
-        # the requested datetime is between their start and end datetimes
         if "eq" in datetime_search:
-            should.extend(
-                [
-                    Q(
-                        "bool",
-                        filter=[
-                            Q(
-                                "term",
-                                properties__datetime=datetime_search["eq"],
-                            ),
-                        ],
-                    ),
-                    Q(
-                        "bool",
-                        filter=[
-                            Q(
-                                "range",
-                                properties__start_datetime={
-                                    "lte": datetime_search["eq"],
-                                },
-                            ),
-                            Q(
-                                "range",
-                                properties__end_datetime={
-                                    "gte": datetime_search["eq"],
-                                },
-                            ),
-                        ],
-                    ),
-                ]
-            )
-
-        # If the request is a date range return
-        # items with datetimes within the requested date range OR
-        # their startdatetime ithin the requested date range OR
-        # their enddatetime ithin the requested date range OR
-        # the requested daterange within their start and end datetimes
+            # For exact matches, include:
+            # 1. Items with matching exact datetime
+            # 2. Items with datetime:null where the time falls within their range
+            should = [
+                Q(
+                    "bool",
+                    filter=[
+                        Q("exists", field="properties.datetime"),
+                        Q("term", **{"properties__datetime": datetime_search["eq"]}),
+                    ],
+                ),
+                Q(
+                    "bool",
+                    must_not=[Q("exists", field="properties.datetime")],
```

Comment:
The current [best practices](https://github.com/radiantearth/stac-api-spec/blob/release/v1.0.0/stac-spec/best-practices.md#datetime-selection) recommends that you populate `datetime` even if you have a date range.

_"The specification does allow one to set the datetime field to null, but it is strongly recommended to populate the single datetime field, as that is what many clients will search on. If it is at all possible to pick a nominal or representative datetime then that should be used."_

So we should probably loosen the search (remove this line?) or update the recommended practice.

Reply:
@rhysrevans3 Can you look at this issue #396? I am not 100% sure on what the right approach should be.

Reply:
I think that for some items, defined by start and end datetimes, it may not make sense to set a datetime value just for the sake of doing so. The stac spec itself allows null datetime values.

Reply:
I like the idea of having an env var to allow it to be configurable if the start and end datetimes will be included in search if the datetime is set. Do we a want to have a way of signalling to users what the expected behaviour of the API is?

Reply:
We could have a few options; default - this one maybe, start and end datetimes included, only datetime

Reply:
I'll open an issue for this so we can work on it later. We can definitely use logging too.